dotnet-efcore/EntityFrameworkCoreVsDapper/EntityFrameworkCoreVsDapper/Repositories/PersonsRepository.cs

```diff
@@ -0,0 +1,64 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using Dapper;
+using EntityFrameworkCoreVsDapper.EFCore;
+using EntityFrameworkCoreVsDapper.Models;
+using System.Configuration;
+using System.Data.Common;
+using System.Data.SqlClient;
+
+namespace EntityFrameworkCoreVsDapper.Repositories
+{
+    [Orderer(BenchmarkDotNet.Order.SummaryOrderPolicy.FastestToSlowest)]
+    [GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByParams)]
+    public class PersonsRepository
+    {
+        private SqlConnection _dbConnection;
+        private PersonsContext _context;
+
+        public PersonsRepository() 
+        {
+            Setup();
+        }
+
+        [IterationSetup]
+        public void Setup()
+        {
+            _dbConnection = new SqlConnection(ConfigurationManager.ConnectionStrings["PersonsDB"].ConnectionString);
+            _context = new PersonsContext();
+        }
+
+        [Benchmark]
+        [Arguments(10000)]
+        [Arguments(100000)]
+        [Arguments(1000000)]
+        public List<Person> GetXPersonsDapper(int personsNumber)
+        {
+            using (_dbConnection)
+            {
+                _dbConnection.Open();
+
+                return _dbConnection.Query<Person>($"SELECT TOP({personsNumber}) * FROM Persons").ToList();
+            }
+        }
+
+        [Benchmark]
+        [Arguments(10000)]
+        [Arguments(100000)]
+        [Arguments(1000000)]
+        public List<Person> GetXPersonsEFCore(int personsNumber)
+        {
+            using (_context)
```

Comment:
This using needs to be removed as we are disposing of the context in the `Cleanup` method. Also, when using this class outside of a benchmark context, we could run into use after dispose issues. As a general rule, if we are not the one creating the object, we should not be the one to dispose of it. `_context` being a member variable indicates that our class should be made IDisposable and implement the Dispose pattern.